src/js/language_label.js

```diff
@@ -1,15 +1,176 @@
+langHints.textContent = "Begin typing to add languages";
+
+var langCancel = labelControlElement("button", "language-cancel");
+langCancel.textContent = "X";
+Object.assign(langCancel.style, {
+  "margin-top": "0.3em",
+});
+
+langHeader.appendChild(langCancel);
+langHeader.appendChild(langHints);
+
+hide(langHeader);
+
+function hide(element) {
+  Object.assign(element.style, {
+    display: "none",
+  });
+}
+function show(element) {
+  element.style.removeProperty("display");
+}
+
+var langCodes = [
+  { id: "am", name: "Amharic" },
+  { id: "ar", name: "Arabic" },
+  { id: "az", name: "Azerbaijani" },
+  { id: "be", name: "Belarusian" },
+  { id: "bg", name: "Bulgarian" },
+  { id: "br", name: "Breton" },
+  { id: "bs", name: "Bosnian" },
+  { id: "ca", name: "Catalan; Valencian" },
+  { id: "co", name: "Corsican" },
+  { id: "cs", name: "Czech" },
+  { id: "cy", name: "Welsh" },
+  { id: "da", name: "Danish" },
+  { id: "de", name: "German" },
+  { id: "el", name: "Greek, Modern" },
+  { id: "en", name: "English" },
+  { id: "eo", name: "Esperanto" },
+  { id: "es", name: "Spanish" },
+  { id: "et", name: "Estonian" },
+  { id: "eu", name: "Basque" },
+  { id: "fi", name: "Finnish" },
+  { id: "fr", name: "French" },
+  { id: "fy", name: "Western Frisian" },
+  { id: "ga", name: "Irish" },
+  { id: "gd", name: "Gaelic" },
+  { id: "he", name: "Hebrew" },
+  { id: "hi", name: "Hindi" },
+  { id: "hr", name: "Croatian" },
+  { id: "hu", name: "Hungarian" },
+  { id: "hy", name: "Armenian" },
+  { id: "id", name: "Indonesian" },
+  { id: "is", name: "Icelandic" },
+  { id: "it", name: "Italian" },
+  { id: "ja", name: "Japanese" },
+  { id: "ja_kana", name: "Japanese Kana" },
+  { id: "ja-Latn", name: "Japanese Romanized" },
+  { id: "ja-Hira", name: "Japanese Hiragana" },
+  { id: "ka", name: "Georgian" },
+  { id: "kk", name: "Kazakh" },
+  { id: "kn", name: "Kannada" },
+  { id: "ko", name: "Korean" },
+  { id: "ko-Latn", name: "Korean Romanized" },
+  { id: "ku", name: "Kurdish" },
+  { id: "la", name: "Latin" },
+  { id: "lb", name: "Luxembourgish" },
+  { id: "lt", name: "Lithuanian" },
+  { id: "lv", name: "Latvian" },
+  { id: "mk", name: "Macedonian" },
+  { id: "mt", name: "Maltese" },
+  { id: "ml", name: "Malayalam" },
+  { id: "nl", name: "Dutch" },
+  { id: "no", name: "Norwegian" },
+  { id: "oc", name: "Occitan" },
+  { id: "pl", name: "Polish" },
+  { id: "pt", name: "Portuguese" },
+  { id: "rm", name: "Romansh" },
+  { id: "ro", name: "Romanian" },
+  { id: "ru", name: "Russian" },
+  { id: "sk", name: "Slovak" },
+  { id: "sl", name: "Slovenian" },
+  { id: "sq", name: "Albanian" },
+  { id: "sr", name: "Serbian" },
+  { id: "sr-Latn", name: "Serbian Romanized" },
+  { id: "sv", name: "Swedish" },
+  { id: "ta", name: "Tamil" },
+  { id: "te", name: "Telugu" },
+  { id: "th", name: "Thai" },
+  { id: "tr", name: "Turkish" },
+  { id: "uk", name: "Ukrainian" },
+  { id: "zh", name: "Chinese" },
+];
```

Comment:
As with https://github.com/ZeLonewolf/openstreetmap-americana/pull/609#discussion_r1044150817, use `Intl.DisplayNames` to format these locale codes. Most of the codes conform to the IETF’s BCP&nbsp;47 standard, so they’ll localize correctly with the following code:

```suggestion
let languageNames = new Intl.DisplayNames(locales, { type: "language" });
let langCodes = [
  "am",
  "ar",
  "az",
  "be",
  "bg",
  "br",
  "bs",
  "ca",
  "co",
  "cs",
  "cy",
  "da",
  "de",
  "el",
  "en",
  "eo",
  "es",
  "et",
  "eu",
  "fi",
  "fr",
  "fy",
  "ga",
  "gd",
  "he",
  "hi",
  "hr",
  "hu",
  "hy",
  "id",
  "is",
  "it",
  "ja",
  // "ja_kana",
  "ja-Latn",
  "ja-Hira",
  "ka",
  "kk",
  "kn",
  "ko",
  "ko-Latn",
  "ku",
  "la",
  "lb",
  "lt",
  "lv",
  "mk",
  "mt",
  "ml",
  "nl",
  "no",
  "oc",
  "pl",
  "pt",
  "rm",
  "ro",
  "ru",
  "sk",
  "sl",
  "sq",
  "sr",
  "sr-Latn",
  "sv",
  "ta",
  "te",
  "th",
  "tr",
  "uk",
  "zh",
].map((locale) => languageNames.of(locale));
```

Reply:
The only nonstandard code is `ja_kana`, but `name:ja_kana` has long been deprecated in favor of `ja-Hira` (which is already in the list). If we want to use `name:ja_kana` when the user requests `ja-Hrkt` (hiragana+katakana), then the translation would happen in `getLocalizedNameExpression()` in label.js.